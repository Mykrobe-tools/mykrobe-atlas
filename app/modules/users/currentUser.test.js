/* @flow */

import sagaHelper from 'redux-saga-testing';

import {
  createCurrentUser,
  requestCurrentUser,
  updateCurrentUser,
  deleteCurrentUser,
  createCurrentUserWorker,
  requestCurrentUserWorker,
  updateCurrentUserWorker,
  deleteCurrentUserWorker,
} from './currentUser';

const payload = { test: true };

describe('user module should create user', () => {
  const test = sagaHelper(createCurrentUserWorker(createCurrentUser(payload)));
  test('execute request', result => {
    expect(result).toMatchSnapshot();
    return { success: true };
  });
});

describe('user module should request user', () => {
  const test = sagaHelper(requestCurrentUserWorker(requestCurrentUser()));
  test('execute request', result => {
    expect(result).toMatchSnapshot();
  });
});

describe('user module should update user', () => {
  const test = sagaHelper(updateCurrentUserWorker(updateCurrentUser(payload)));
  test('execute request', result => {
    expect(result).toMatchSnapshot();
  });
});

describe('user module should delete user', () => {
  const test = sagaHelper(deleteCurrentUserWorker(deleteCurrentUser()));
  test('execute request', result => {
    expect(result).toMatchSnapshot();
    return { success: true };
  });
});
