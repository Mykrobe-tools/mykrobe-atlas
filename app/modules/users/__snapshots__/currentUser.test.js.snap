// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user module should create user execute request 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "action": Object {
          "meta": undefined,
          "payload": Object {
            "test": true,
          },
          "type": "users/currentUser/CREATE_CURRENT_USER",
        },
        "crudOptions": Object {
          "onSuccess": [Function],
          "operationId": "usersCreate",
        },
        "meta": undefined,
        "parameters": Object {
          "test": true,
        },
        "types": Array [
          "users/currentUser/CREATE_CURRENT_USER_REQUEST",
          "users/currentUser/CREATE_CURRENT_USER_SUCCESS",
          "users/currentUser/CREATE_CURRENT_USER_FAILURE",
          "users/currentUser/CREATE_CURRENT_USER_ABORT",
          "users/currentUser/CREATE_CURRENT_USER_UPLOAD_PROGRESS",
          "users/currentUser/CREATE_CURRENT_USER_DOWNLOAD_PROGRESS",
        ],
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`user module should delete user execute request 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "action": Object {
          "meta": undefined,
          "payload": undefined,
          "type": "users/currentUser/DELETE_CURRENT_USER",
        },
        "crudOptions": Object {
          "onSuccess": [Function],
          "operationId": "currentUserDelete",
        },
        "meta": undefined,
        "parameters": undefined,
        "types": Array [
          "users/currentUser/DELETE_CURRENT_USER_REQUEST",
          "users/currentUser/DELETE_CURRENT_USER_SUCCESS",
          "users/currentUser/DELETE_CURRENT_USER_FAILURE",
          "users/currentUser/DELETE_CURRENT_USER_ABORT",
          "users/currentUser/DELETE_CURRENT_USER_UPLOAD_PROGRESS",
          "users/currentUser/DELETE_CURRENT_USER_DOWNLOAD_PROGRESS",
        ],
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`user module should request user execute request 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "action": Object {
          "meta": undefined,
          "payload": undefined,
          "type": "users/currentUser/REQUEST_CURRENT_USER",
        },
        "crudOptions": Object {
          "onFailure": [Function],
          "operationId": "currentUserGet",
        },
        "meta": undefined,
        "parameters": undefined,
        "shouldExecuteJsonApiAction": undefined,
        "types": Array [
          "users/currentUser/REQUEST_CURRENT_USER_REQUEST",
          "users/currentUser/REQUEST_CURRENT_USER_SUCCESS",
          "users/currentUser/REQUEST_CURRENT_USER_FAILURE",
          "users/currentUser/REQUEST_CURRENT_USER_ABORT",
          "users/currentUser/REQUEST_CURRENT_USER_UPLOAD_PROGRESS",
          "users/currentUser/REQUEST_CURRENT_USER_DOWNLOAD_PROGRESS",
        ],
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`user module should update user execute request 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "action": Object {
          "meta": undefined,
          "payload": Object {
            "test": true,
          },
          "type": "users/currentUser/UPDATE_CURRENT_USER",
        },
        "crudOptions": Object {
          "onSuccess": [Function],
          "operationId": "currentUserUpdate",
        },
        "meta": undefined,
        "parameters": Object {
          "test": true,
        },
        "types": Array [
          "users/currentUser/UPDATE_CURRENT_USER_REQUEST",
          "users/currentUser/UPDATE_CURRENT_USER_SUCCESS",
          "users/currentUser/UPDATE_CURRENT_USER_FAILURE",
          "users/currentUser/UPDATE_CURRENT_USER_ABORT",
          "users/currentUser/UPDATE_CURRENT_USER_UPLOAD_PROGRESS",
          "users/currentUser/UPDATE_CURRENT_USER_DOWNLOAD_PROGRESS",
        ],
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
